controller.go

```diff
@@ -157,21 +156,33 @@ func (c *Controller) handleObject(obj interface{}) {
 		logger.V(4).Info("Recovered deleted object", "resourceName", object.GetName())
 	}
 
-	logger.V(4).Info("Processing object", "object", klog.KObj(object))
-	if ownerRef := metav1.GetControllerOf(object); ownerRef != nil {
-		// If this object is not owned by a MachineLearningAlgorithm, skip it
-		if ownerRef.Kind != "MachineLearningAlgorithm" {
-			return
+	switch ot := object.(type) {
```

Comment:
```suggestion
	switch object.(type) {
```


I think we can remove the assignment, and hardcode the type value in `HandleErrorWithContext` since it will always be of type `v1.MachineLearningAlgorithm`